flowapi/flowapi/user_model.py

```diff
@@ -0,0 +1,285 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from json import JSONEncoder
+
+from flask_jwt_extended.exceptions import UserClaimsVerificationError
+from typing import Dict, Union, List
+
+from flask_jwt_extended import get_jwt_claims, get_jwt_identity
+from quart import current_app, request
+
+
+class UserObject:
+    """
+    Class to represent a user's permissions as loaded from a JWT.
+    Provided methods which check a user has access to perform specific actions
+    against API endpoints.
+
+    Parameters
+    ----------
+    username : str
+        Name of the user
+    claims : dict
+        Dictionary giving a whitelist of the user's claims
+    """
+
+    def __init__(
+        self,
+        username: str,
+        claims: Dict[str, Dict[str, Union[List[str], Dict[str, bool]]]],
+    ) -> None:
+        self.username = username
+        self.claims = claims
+
+    def has_access(
+        self, *, action: str, query_kind: str, aggregation_unit: str
+    ) -> bool:
+        """
+        Returns true if the user can do 'action' with this kind of query at this unit of aggregation.
+
+        Parameters
+        ----------
+        action: {'run', 'poll', 'get_results'}
+            Action to check
+        query_kind : str
+            Kind of the query
+        aggregation_unit : str
+            Aggregation unit/level of resolution
+
+        Returns
+        -------
+        bool
+            True if the user can do 'action' with this query
+
+        Raises
+        ------
+        UserClaimsVerificationError
+            If the user cannot do action with this kind of query at this level of aggregation
+        """
+        try:
+            action_rights = self.claims[query_kind]["permissions"][action]
+            unit_rights = (
```

Comment:
Nitpick, but how about calling this `aggregation_rights` instead of `unit_right`?